corehq/apps/data_cleaning/tasks.py

```diff
@@ -116,13 +116,27 @@ def _prune_records_and_complete_session(session, errored_doc_ids):
     session.save(update_fields=['completed_on'])
 
 
-def _create_case_blocks(session, records):
+def _create_case_blocks(session, records, errored_doc_ids):
     blocks = []
     case_ids = [rec.doc_id for rec in records]
     cases = {c.case_id: c for c in CommCareCase.objects.get_cases(case_ids, session.domain)}
     for record in records:
         case = cases[record.doc_id]
-        update = record.get_edited_case_properties(case)
+        try:
+            update = record.get_edited_case_properties(case)
+        except Exception as e:
+            errored_doc_ids.append(record.doc_id)
```

Comment:
might make sense to extract this into a utility method to ease readability

Reply:
done https://github.com/dimagi/commcare-hq/pull/36358/commits/6c67945a96abc7bad7b14e259eea42a62fa67652